### DISCUSSION TITLE
Rename/Rewriting /api to /v2, why is it not working?

### DISCUSSION THREAD

**Discussion by Wedz0ff** [vercel/next.js#38185] [Help] [answered]
I'm facing a problem that I can't solve, my API files on next.js are located on /src/pages/api/

In any case, rewriting is not working, i'm trying to rewrite baseUrl/api/ to baseUrl/v2, here's the next.config.js code:

```js 
module.exports = {
  async rewrites() {
    return [
      {
        source: "/api/:slug*",
        destination: "/v2/:slug*",
      },
    ];
  },
};
```
I've read about middlewares on documentation, but couldn't have a way to do that, is it possible to do on next.js or should I just migrate my project to Express.js?

**Comment by icyJoseph** [comment, marked as answer]
Hi, you have to do it like:

```js
module.exports = {
  async rewrites() {
    return [
      {
        source: "/v2/:slug*",
        destination: "/api/:slug*",
      },
    ];
  },
};
```

The querying entity, browser, curl, postman etc, will see, `/v2/:slug*`, but Next.js serves the contents of `/api/:slug*`.

For example, if I have `/api/hello`, inside pages, in the browser I can do `http://localhost:3000/v2/hello`, and then I see the contents of `/api/hello`.

**Reply by Wedz0ff** [reply]
Thank you, now I realise it was a really rookie mistake, thank you again!
